torch/nested/_nested_tensor.py

```diff
@@ -0,0 +1,180 @@
+import unittest
+
+import torch
+from torch import Tensor, SymInt
+from torch.autograd import Function
+from torch.testing._internal.common_utils import run_tests, TestCase
+from torch.utils._pytree import tree_map
+from typing import List, Union
+from functools import reduce
+import operator
+
+# Python tensor subclass version of NT.
+# Properties:
+#  * Constrained to contiguous-only
+#  * Returned for fake / meta conversion
+#  * Supports size() / stride() -> [SymInt] for dynamic shapes
+#  * Supports size() / stride() -> [Union[int, List[int]]] for non-dynamic shapes
+class NestedTensor(Tensor):
+    @staticmethod
+    def __new__(cls, buffer, *args, **kwargs):
+        # Put in obvious garbage; we don't ever want to actually hit the C++ side of this thing
+        return Tensor._make_subclass(cls, torch.empty(0))
+
+    # sizes should be a list of SymInts for the dynamic shapes case
+    def __init__(self, buffer: Tensor, sizes: List[Union[int, List[int], SymInt]]):
+        self.buffer = buffer
```

Comment:
doc for buffer format would be handy for readers

Reply:
You don't want the ragged sizes to be a List[int] right? Isn't it going to be a tensor?